Orange/widgets/unsupervised/owhierarchicalclustering.py

```diff
@@ -962,11 +965,15 @@ def axis_view(orientation):
     @Inputs.distances
     def set_distances(self, matrix):
         self.error()
+        self.Error.clear()
         if matrix is not None:
             N, _ = matrix.shape
             if N < 2:
                 self.error("Empty distance matrix")
                 matrix = None
+            if not np.all(np.isfinite(matrix)):
```

Comment:
No, not like that. If it comes here, `matrix` can be set to `None` in the line above. Another `if matrix is not None` is needed here.